server/src/main/java/org/elasticsearch/action/ActionModule.java

```diff
@@ -436,6 +440,15 @@ public ActionModule(Settings settings, IndexNameExpressionResolver indexNameExpr
                     throw new IllegalArgumentException("Cannot have more than one plugin implementing a REST wrapper");
                 }
                 restWrapper = newRestWrapper;
+                if (restWrapper.getClass().getCanonicalName() == null ||
+                    restWrapper.getClass().getCanonicalName().startsWith("org.elasticsearch") == false) {
+                    deprecationLogger.deprecate(DeprecationCategory.SECURITY, "3rd_party_rest_deprecation", "The " +
+                        plugin.getClass().getName() + "plugin installs a custom REST wrapper. This functionality is deprecated and will " +
+                        "not be possible in Elasticsearch 8.0. If this plugin is intended to provide security features for Elasticsearch " +
+                        "then you should switch to using the built-in Elasticsearch features instead. " +
+                        "Read https://www.elastic.co/guide/en/elasticsearch/reference/" + Version.CURRENT.major + "." +
```

Comment:
We have avoided putting URLs in deprecation messages as they break too easily. In fact, I cannot find any other occurrence of one. Can we remove this?